From 58eab99e6b5fbcd7c9bb4f9a5f0bd5feb0b3edf3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20Br=C3=BCns?= <stefan.bruens@rwth-aachen.de>
Date: Wed, 12 Aug 2020 18:27:23 +0200
Subject: [PATCH] Replace KLineEdit with QLineEdit

The password field uses none of the KLineEdit features like completion,
so QLineEdit suffices.
---
 CMakeLists.txt      | 1 -
 krfb/CMakeLists.txt | 1 -
 krfb/mainwindow.cpp | 3 +--
 krfb/mainwindow.h   | 4 ++--
 4 files changed, 3 insertions(+), 6 deletions(-)

Index: krfb-20.08.2/CMakeLists.txt
===================================================================
--- krfb-20.08.2.orig/CMakeLists.txt	2020-10-09 23:46:03.405438902 +0200
+++ krfb-20.08.2/CMakeLists.txt	2020-10-09 23:46:05.847580539 +0200
@@ -30,7 +30,6 @@
 
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS
     I18n
-    Completion
     Config
     CoreAddons
     Crash
Index: krfb-20.08.2/krfb/CMakeLists.txt
===================================================================
--- krfb-20.08.2.orig/krfb/CMakeLists.txt	2020-10-09 23:46:03.405438902 +0200
+++ krfb-20.08.2/krfb/CMakeLists.txt	2020-10-09 23:46:05.847580539 +0200
@@ -101,7 +101,6 @@
                        ${X11_X11_LIB}
                        ${X11_Xdamage_LIB}
                        Qt5::Network
-                       KF5::Completion
                        KF5::CoreAddons
                        KF5::DBusAddons
                        KF5::DNSSD
Index: krfb-20.08.2/krfb/mainwindow.cpp
===================================================================
--- krfb-20.08.2.orig/krfb/mainwindow.cpp	2020-10-05 07:07:12.000000000 +0200
+++ krfb-20.08.2/krfb/mainwindow.cpp	2020-10-09 23:46:05.847580539 +0200
@@ -21,7 +21,6 @@
 #include <KMessageWidget>
 #include <KStandardAction>
 #include <KActionCollection>
-#include <KLineEdit>
 #include <KNewPasswordDialog>
 #include <KPluginLoader>
 #include <KPluginMetaData>
@@ -107,7 +106,7 @@
     setAttribute(Qt::WA_DeleteOnClose, false);
 
     m_passwordEditable = false;
-    m_passwordLineEdit = new KLineEdit(this);
+    m_passwordLineEdit = new QLineEdit(this);
     m_passwordLineEdit->setVisible(false);
     m_passwordLineEdit->setAlignment(Qt::AlignHCenter);
 
Index: krfb-20.08.2/krfb/mainwindow.h
===================================================================
--- krfb-20.08.2.orig/krfb/mainwindow.h	2020-10-05 07:07:12.000000000 +0200
+++ krfb-20.08.2/krfb/mainwindow.h	2020-10-09 23:46:05.847580539 +0200
@@ -15,7 +15,7 @@
 
 #include <KXmlGuiWindow>
 
-class KLineEdit;
+class QLineEdit;
 
 class MainWindow : public KXmlGuiWindow
 {
@@ -43,7 +43,7 @@
     private:
         Ui::MainWidget m_ui;
         bool m_passwordEditable;
-        KLineEdit *m_passwordLineEdit;
+        QLineEdit *m_passwordLineEdit;
 };
 
 #endif
